src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -361,29 +328,31 @@
     (let [population-registry (get-in metrics/metrics-registries [:population :registry])]
       (pop/initialize-population-metrics! population-registry read-db))
 
-    (when (.exists discard-dir)
-      (dlo/create-metrics-for-dlo! discard-dir))
-    (let [broker (try
-                   (log/info "Starting broker")
-                   (mq/build-and-start-broker! "localhost"
-                                               (conf/mq-dir config)
-                                               command-processing)
-                   (catch java.io.EOFException e
-                     (log/error
-                      "EOF Exception caught during broker start, this "
-                      "might be due to KahaDB corruption. Consult the "
-                      "PuppetDB troubleshooting guide.")
-                     (throw e)))
+    ;; Error handling here?
+    (let [stockdir (conf/stockpile-dir config)
+          command-chan (async/chan
+                         (if (:catalog-facts-bash command-processing)
```

Comment:
This config option enables "bash in place" where catalogs/factsets for a given certname are marked as "delete" when a new catalog/factset from the same certname is enqueued. The new catalog/fact is put at the tail of the queue and won't be processed sooner, but the old catalogs/factsets won't be process, just deleted from disk when they are at the head of the queue